init/cli/start.go

```diff
@@ -15,48 +15,37 @@
 package cli
 
 import (
-	"os"
-
 	"github.com/palantir/pkg/cli"
 	"github.com/pkg/errors"
 
 	"github.com/palantir/go-java-launcher/init/lib"
-	"github.com/palantir/go-java-launcher/launchlib"
 )
 
 func startCommand() cli.Command {
 	return cli.Command{
 		Name: "start",
 		Usage: `
-Launches the process defined by the static and custom configurations at service/bin/launcher-static.yml and
-var/conf/launcher-custom.yml. Writes its PID to var/run/service.pid and redirects its output to var/log/startup.log. If
-successful, exits 0, otherwise exits 1 and writes an error message to stderr.`,
+Ensures the service defined by the static and custom configurations at service/bin/launcher-static.yml and
+var/conf/launcher-custom.yml is running and its outputs are redirecting to var/log/startup.log and other
+var/log/${SUB_PROCESS}-startup.log files. If successful, exits 0, otherwise exits 1 and writes an error message to
+stderr.`,
 		Action: func(_ cli.Context) error {
 			return start()
 		},
 	}
 }
 
 func start() error {
```

Comment:
I find the new logical flow of this function overall very convoluted.  We currently:
- try and load the launcher config then the pid file to determine processes which aren't running
- if that fails, reload the launcher config and return everything
- start not running services
- load the pid file again
- add the new pids to the deserialized pid file
- save the pid file

Why are we:
- Load config
- Load pid file to determine which of config are already running
- Start the rest
- Save pids we knew were previously running + the ones we started, back to the pid file

This seems much more direct, and avoids constantly reserializing files from disk.